net/service/handlers.go

```diff
@@ -1486,6 +1558,102 @@ func (service *OpenBazaarService) handleStore(pid peer.ID, pmes *pb.Message, opt
 	return m, nil
 }
 
+func (service *OpenBazaarService) handleOrderPayment(peer peer.ID, pmes *pb.Message, options interface{}) (*pb.Message, error) {
+	// Unmarshall
+	if pmes.Payload == nil {
+		return nil, errors.New("payload is nil")
+	}
+	paymentDetails := new(pb.OrderPaymentTxn)
+	err := ptypes.UnmarshalAny(pmes.Payload, paymentDetails)
+	if err != nil {
+		return nil, err
+	}
+
+	wal, err := service.node.Multiwallet.WalletForCurrencyCode(paymentDetails.GetCoin())
+	if err != nil {
+		return nil, err
+	}
+
+	wal0, ok := wal.(wallet.WalletMustManuallyAssociateTransactionToOrder)
+
+	if !ok {
+		return nil, nil
+	}
+
+	chash, err := chainhash.NewHashFromStr(paymentDetails.GetTransactionID())
+	if err != nil {
+		return nil, err
+	}
+
+	txn, err := wal.GetTransaction(*chash)
+	if err != nil {
+		return nil, err
+	}
+
+	contract, _, _, _, _, _, err := service.datastore.Sales().GetByOrderId(paymentDetails.OrderID)
+	if err != nil {
+		return nil, net.OutOfOrderMessage
+	}
+
+	if contract.VendorOrderConfirmation != nil &&
+		contract.BuyerOrder.Payment.Method != pb.Order_Payment_MODERATED {
+
+		// the seller has confirmed the order, so a simple check of
+		// the addresses and we are good to proceed
+		if !u.AreAddressesEqual(contract.VendorOrderConfirmation.PaymentAddress, txn.ToAddress) {
+			log.Debugf("mismatched payment address details: orderID: %s, expectedAddr: %s, actualAddr: %s",
+				paymentDetails.OrderID, contract.VendorOrderConfirmation.PaymentAddress, txn.ToAddress)
```

Comment:
Well if you are asking for a use case/scenario, then I have none. It is a check I put in to double check before proceeding. In all probability of this happening is quite small since orderID is used for creating the ORDER_PAYMENT message in the first place and the code would not reach here if the order was not found.
 The only real recovery is for the buyer to realize that the payment has been made to another address which does not match the address specified in the contract.

Reply:
@amangale will raise with @drwasho what follow up work needs to be done when the order lands in this state... currently we don't recover from this case and would be nice to have a plan before resolving this PR.